### PR TITLE
chore: add GCI linter to group import statements

### PR DESCRIPTION
**Pull request by miparnisari**
Because I want to introduce consistency in the grouping of imports, e.g. this feels slightly messy:

image

versus with this PR:
image

### CODE REVIEW

File: .golangci.yaml [hunk 1]
```diff
@@ -52,13 +53,20 @@ linters-settings:
...
     alias:
       - pkg: github.com/openfga/api/proto/openfga/v1
         alias: openfgav1
+  gci:
+    sections:
+      - standard # Standard section: captures all standard packages.
+      - default # Default section: contains all imports that could not be matched to another section type.
+      - prefix(github.com/openfga/api)  # Custom section: groups all imports with the specified Prefix.
+      - prefix(github.com/openfga/openfga)
+      - prefix(github.com/openfga/language)
```
Comment (DanCech) [inline .golangci.yaml:69]: interesting, shouldn't these 2 be swapped based on order of the actual imports in the files?

Comment (miparnisari) [reply .golangci.yaml:69]: Yeah, i noticed after sending this PR that either the tool doesn't work as I expect, or I don't understand how to configure it properly.

Comment (miparnisari) [reply .golangci.yaml]: Removing `local-prefixes: "github.com/openfga/openfga"` seems to have fixed this.

**Review by jpadilla — DISMISSED** [review]

File: .golangci.yaml (hunk 1 above)
Comment (DanCech) [reply .golangci.yaml]: Not sure that we want to remove local-prefixes, I don't know what if anything goimports uses that for

Right now we're mixing local imports with other repos in the openfga org, which is weird imho. The standard approach would be that those other repos are considered part of the external group.

I'd suggest either going back to the standard 3-group layout; `standard, default, localmodule` or a 4-group layout; `standard, default, prefix(github.com/openfga), localmodule`

Comment (miparnisari) [reply .golangci.yaml]: goimports' local-prefixes seems to affect the order of the imports: https://golangci-lint.run/usage/linters/#goimports

> Reformat imports in autofix mode.

image

> Right now we're mixing local imports with other repos in the openfga org, which is weird imho.

do you prefer to have

`openfga/api`, then
`openfga/language`, then
`openfga/openfga`?

Comment (DanCech) [reply .golangci.yaml]: this config seems to do that:
```
  goimports:
    local-prefixes: "github.com/openfga/openfga"
  gci:
    sections:
      - standard # Standard section: captures all standard packages.
      - default # Default section: contains all imports that could not be matched to another section type.
      - prefix(github.com/openfga)  # Custom section: groups all imports with the specified Prefix.
      - localmodule  # Local section: contains all imports that are part of the same module.
```

Comment (miparnisari) [reply .golangci.yaml]: Added `localmodule`.

**Review by DanCech — APPROVED** [review]
Looks great!

**Review by jpadilla — APPROVED** [review]

**Pull request merged**